packages/material-ui/src/Paper/Paper.test.js

```diff
@@ -23,63 +23,44 @@ describe('<Paper />', () => {
 
   describe('prop: square', () => {
     it('can disable the rounded class', () => {
-      const { getByTestId } = render(
-        <Paper data-testid="root" square>
-          Hello World
-        </Paper>,
-      );
-
-      expect(getByTestId('root')).not.to.have.class(classes.rounded);
+      const { container } = render(<Paper square>Hello World</Paper>);
+      expect(container.firstChild).not.to.have.class(classes.rounded);
```

Comment:
`getByTestId` is the correct choice. `container.firstChild` is acceptable for faster iteration but more brittle. So when we already have the robust version we don't need to downgrade to the brittle version.

Reply:
Why do we need a more robust version? Which problem in the future is it meant to edge against?

I think that using `container.firstChild` puts a healthy constrain in place, as it makes it more painful to have components that host multiple DOM nodes.

Reply:
I have spotted this opportunity looking at the test changes of #21748.

Reply:
`expect(container.firstChild)` has 163 macthes in the codebase, `getByTestId('root')` has 34.

Reply:
> Why do we need a more robust version? Which problem in the future is it meant to edge against?

We should be able to add arbitrary wrappers in the client render whose implementation is is irrelevant for the test.

> I think that using container.firstChild puts a healthy constrain in place,

Could you explain why this is "healthy"?

> expect(container.firstChild) has 163 matches in the codebase, getByTestId('root') has 34.

What does this tell us?

Reply:
> Why do we need a more robust version? 

We don't need to **add** which why my premise. I won't ever reject that. But if we already have the robust version in place we should not **remove** it.

Reply:
> We should be able to add arbitrary wrappers in the client render whose implementation is is irrelevant for the test.

Ok, I guess I can't imagine a case where we would need to add a div wrapper.

> Could you explain why this is "healthy"?

I was trying to optimize for avoiding cases like #19613, but I think that you are right, most of the value probably comes from making tests easier to write which, without, makes it harder to have more & better (increase coverage).

> What does this tell us?

1. We have, so far, valued the ease of writing tests.
2. Being able to add a DOM node in the createClientRender would require quite some refactoring, maybe it will never worth it?